Library/ViewModels/ProjectPamphletCreatorHeaderCellViewModelTests.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+@testable import KsApi
+@testable import Library
+import Prelude
+import ReactiveExtensions_TestHelpers
+
+final class ProjectPamphletCreatorHeaderCellViewModelTests: TestCase {
+  private let vm: ProjectPamphletCreatorHeaderCellViewModelType = ProjectPamphletCreatorHeaderCellViewModel()
+
+  private let buttonTitle = TestObserver<String, Never>()
+  private let launchDateLabelAttributedText = TestObserver<String, Never>()
+  private let notifyDelegateViewProgressButtonTapped = TestObserver<Project, Never>()
+
+  override func setUp() {
+    super.setUp()
+
+    self.vm.outputs.buttonTitle.observe(self.buttonTitle.observer)
+    self.vm.outputs.launchDateLabelAttributedText.map { $0.string }
+      .observe(self.launchDateLabelAttributedText.observer)
+    self.vm.outputs.notifyDelegateViewProgressButtonTapped.observe(
+      self.notifyDelegateViewProgressButtonTapped.observer
+    )
+  }
+
+  func testButtonTitle_LiveProject() {
+    let project = Project.template
+      |> \.state .~ .live
+
+    self.buttonTitle.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: project)
+
+    self.buttonTitle.assertValue("View progress")
+  }
+
+  func testButtonTitle_NonLiveProject() {
+    let project = Project.template
+      |> \.state .~ .successful
+
+    self.buttonTitle.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: project)
+
+    self.buttonTitle.assertValue("View dashboard")
+  }
+
+  func testLaunchDateLabelAttributedText() {
+    let project = Project.template
```

Comment:
Can we set the `.launchedAt` variable on the project to a date we've defined so we can see that it's actually displaying the correct date? Otherwise this test feels a little hand-wavey.

Reply:
Thats a good idea!

Reply:
I meant more something like:

```
   // Set the date to Nov 1, 2019
    let dateComponents = DateComponents()
      |> \.month .~ 11
      |> \.day .~ 1
      |> \.year .~ 2_019
      |> \.timeZone .~ TimeZone.init(secondsFromGMT: 0)

      let date = AppEnvironment.current.calendar.date(from: dateComponents)

      let project = Project.template |> \.dates.launchedAt .~ date.timeIntervalSince1970
...
    // Assert that the date is Nov 1, 2019, which is the date you set in the "launchedAt" property
    self.launchDateLabelAttributedText.assertValue("You launched this project on November 1, 2019.")
```
You might also want to set the locale in `withEnvironment` to ensure that the date is always formatted in the same way.

Reply:
Ohh I see!